src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -271,13 +278,40 @@ public static Complex Acos(Complex value)
 
         public static Complex Tan(Complex value)
         {
-            return (Sin(value) / Cos(value));
+            // tan z = sin z / cos z, but to avoid unnecessary repeated trig computations, use
+            //   tan z = \frac{\sin(2x) + i \sinh(2y)}{\cos(2x) + \cosh(2y)}
```

Comment:
I don't find this notation particularly readable in a code comment. Could you re-write it? I think this would be more readable, for example:

```C#
tan(z) = (sin(2x) + i * sinh(2y)) / (cos(2x) + cosh(2y))

// or even

tan(z) = (sin(2x) + i * sinh(2y))
         / (cos(2x) + cosh(2y))
```